packages/lib/server-only/user/delete-user.ts

```diff
@@ -0,0 +1,21 @@
+import { prisma } from '@documenso/prisma';
+
+export const deleteUser = async (name: string) => {
+  const user = await prisma.user.findFirst({
+    where: {
+      name: {
+        contains: name,
+      },
+    },
+  });
+
+  if (!user) {
+    throw new Error(`User with name ${name} not found`);
+  }
+
+  return await prisma.user.delete({
+    where: {
+      id: user.id,
+    },
+  });
```

Comment:
It feels a little dodgy to delete by name due to no unique restraints.

Is there a reason we can't just do something like this?

```
export const deleteUserByEmail = async (email string) => {
  return prisma.user.delete({
    where: {
      email,
    },
  });
});
```

Reply:
It's deleting by ID, not name. Although, I can change it to email if you think it's better.

Reply:
The ID is found by fetching a user by their name though

I think it's better to just delete by unique value, in this case email is unique

Reply:
Ah, I get what you meant. Well, I figured that "test user" is unlikely to clash with other usernames, but I changed it to the email anyways. Thanks!